react/src/base/inputs/SprkDatePicker/SprkDatePicker.js

```diff
@@ -0,0 +1,245 @@
+            month: '2-digit',
+            day: '2-digit',
+            year: 'numeric',
+          })
+          .replace(/[^ -~]/g, ''),
+    };
+
+    if (value) {
+      this.state = {
+        hasValue: true,
+      };
+    } else {
+      this.state = {
+        hasValue: false,
+      };
+    }
+  }
+
+  componentDidMount() {
+    const { datePickerConfig } = this.props;
+    TinyDatePicker(
+      this.datePickerInputRef.current,
+      assign(this.tdpConfig, datePickerConfig),
+    );
+  }
+
+  render() {
+    const {
+      datePickerConfig,
+      variant,
+      additionalClasses,
+      analyticsString,
+      formatter,
+      idString,
+      isDisabled,
+      isValid,
+      ariaDescribedBy,
+      value,
+      id,
+      ...rest
+    } = this.props;
+    const { hasValue } = this.state;
+
+    // Adds class for IE and Edge
+    const handleOnBlur = (e) => {
+      if (e.target.value.length > 0) {
+        this.setState({ hasValue: true });
+      } else {
+        this.setState({ hasValue: false });
+      }
+    };
+
+    return (
+      <input
+        className={classNames('sprk-b-TextInput', additionalClasses, {
+          'sprk-b-TextInput--error': !isValid,
+          'sprk-b-Input--has-floating-label': hasValue && variant === 'huge',
+          'sprk-b-TextInput--has-svg-icon': variant !== 'huge',
+        })}
+        disabled={isDisabled}
+        ref={this.datePickerInputRef}
+        aria-describedby={ariaDescribedBy}
+        data-id={idString}
+        data-analytics={analyticsString}
+        aria-invalid={!isValid}
+        value={isValid && formatter ? formatter(value) : value}
+        onBlur={(e) => handleOnBlur(e)}
+        id={id}
+        {...rest}
+      />
+    );
+  }
+}
+
+SprkDatePicker.propTypes = {
+  /**
+   * Exposes configuration provided
+   * by tiny-date-picker, see [github](https://github.com/chrisdavies/tiny-date-picker) for documentation.
+   */
+  datePickerConfig: PropTypes.shape({
+    /**
+     * What DOM element the Date Picker will be added to. This defaults
+     * to document.body
+     */
+    appendTo: PropTypes.node,
+    /**
+     * Lang can be used to customize the text that is displayed
+     * in the calendar. You can use this to display a different language.
+     */
+    lang: PropTypes.shape({
+      days: PropTypes.arrayOf(PropTypes.string),
+      months: PropTypes.arrayOf(PropTypes.string),
+    }),
+    today: PropTypes.string,
+    clear: PropTypes.string,
+    close: PropTypes.string,
+    /**
+     * format {Date} -> string is a function which takes a
+     * date and returns a string. It can be used to customize
+     * the way a date will look in the input after the user has
+     * selected it, and is particularly useful if you're targeting
+     * a non-US customer.
+     */
+    format: PropTypes.func,
+    /**
+     * parse {string|Date} -> Date is the inverse of format.
+     * If you specify one, you probably should specify the other
```

Comment:
Some of this language could come off weird. Like the "you probably should part". Did you write this? If not, I think it should be replaced with your writing.

Reply:
These were a quick copy/paste from the tiny-date-picker docs.